aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -1425,16 +1426,19 @@ namespace {
     }
   }
 
-  bool transposed(const Tensor& mat) {
+  bool check_valid_strides_and_return_transposed(const Tensor& mat) {
     IntArrayRef tensor_strides = mat.strides();
     IntArrayRef tensor_sizes = mat.sizes();
     int end_dim = mat.dim() - 1;
+    int alignment = 16 / mat.element_size();
```

Comment:
Does it need to be 16 for Bfloat16? I thought the 16 primarily came from 16byte alignment for tma

Reply:
it's for all tma loads, so yes, bf16 is also affected (had the funniest bug when in the test my stride was not divisible by 8, and just got silently wrong results, that's why I put in this check here)

Reply:
Ohh whoops I misread this 8 was what I expected to see and what you have